flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -327,7 +327,11 @@ public CheckpointCoordinator(
 
         try {
             this.checkpointStorageView = checkpointStorage.createCheckpointStorage(job);
-            checkpointStorageView.initializeBaseLocations();
+
+            // directory will not be created if checkpointing is disabled
+            if (baseInterval != Long.MAX_VALUE) {
```

Comment:
I prefer to leverage `isPeriodicCheckpointingConfigured()` method directly, so that we could also avoid the comment.